cardinal/ecs/engine.go

```diff
@@ -759,7 +760,16 @@ func (e *Engine) IsGameLoopRunning() bool {
 	return e.isGameLoopRunning.Load()
 }
 
-func (e *Engine) Shutdown() error {
+func (e *Engine) Shutdown() {
```

Comment:
Shutdown should initiate a graceful shutdown that doesn't return until everything is off, so spinning off a goroutine here isn't appropriate. 

If this works for profiling, and need to be called from a system, why not just call `go e.Shutdown()` from the System?

Reply:
The problem here is if you call Shutdown from within a system. Then Shutdown will block the entire Tick because it's in the system which is in the tick itself. 

Profiling is the thing that necessitated this feature but it's a bit orthogonal.

Reply:
We discussed this with Scott, Jeremy and Tyler. I just won't do this because eCtx is going to be removed.